android/java/org/chromium/chrome/browser/homepage/settings/BraveRadioButtonGroupHomepagePreference.java

```diff
@@ -0,0 +1,109 @@
+import android.widget.RadioGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.preference.PreferenceViewHolder;
+
+import org.chromium.chrome.R;
+import org.chromium.components.browser_ui.widget.RadioButtonWithDescription;
+import org.chromium.components.browser_ui.widget.RadioButtonWithEditText;
+
+/** Brave's version of a radio button group Preference used for Homepage Preference. */
+public final class BraveRadioButtonGroupHomepagePreference
+        extends RadioButtonGroupHomepagePreference {
+    static final String MOBILE_BOOKMARKS_PATH = "chrome-native://bookmarks/folder/1";
+
+    private RadioButtonWithDescription mMobileBookmarks;
+    private RadioButtonWithEditText mCustomUri;
+    private boolean mSelectMobileBookmarks;
+
+    public BraveRadioButtonGroupHomepagePreference(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    /**
+     * Called when the checked radio button has changed. When the selection is cleared, checkedId is
+     * -1.
+     *
+     * @param group The group in which the checked radio button has changed
+     * @param checkedId The unique identifier of the newly checked radio button
+     */
+    @Override
+    public void onCheckedChanged(RadioGroup group, int checkedId) {
+        PreferenceValues preferenceValues = getPreferenceValue();
+        if (preferenceValues == null) return;
+
+        if (checkedId == R.id.brave_homepage_mobile_bookmarks) {
+            maybeSelectMobileBookmarks(true);
+        } else {
+            super.onCheckedChanged(group, checkedId);
+        }
+        maybeClearCustomUri();
+    }
+
+    @Override
+    public void onBindViewHolder(PreferenceViewHolder holder) {
+        super.onBindViewHolder(holder);
+
+        mMobileBookmarks =
+                (RadioButtonWithDescription)
+                        holder.findViewById(R.id.brave_homepage_mobile_bookmarks);
+        assert mMobileBookmarks != null : "Mobile bookmarks button is missing in the layout";
+
+        mCustomUri = (RadioButtonWithEditText) holder.findViewById(R.id.radio_button_uri_edit);
+        assert mCustomUri != null : "Custom URI button missing in the layout";
+
+        maybeSelectMobileBookmarks(false);
+        maybeClearCustomUri();
+    }
+
+    @Override
+    void setupPreferenceValues(@NonNull PreferenceValues value) {
+        super.setupPreferenceValues(value);
+
+        // If it is custom uri with mobile bookmarks path, set the mobile bookmarks radio button
+        // selected.
+        mSelectMobileBookmarks =
+                value.getCheckedOption() == HomepageOption.ENTRY_CUSTOM_URI
+                        && value.getCustomURI().equals(MOBILE_BOOKMARKS_PATH);
+
+        maybeSelectMobileBookmarks(false);
+        maybeClearCustomUri();
+    }
+
+    void maybeSelectMobileBookmarks(boolean forceSelect) {
+        if (mSelectMobileBookmarks || forceSelect) {
+            if (mMobileBookmarks != null) {
```

Comment:
Double `if`s  can be inlined merging the logic into a single statement:

```java

if ((mSelectMobileBookmarks || forceSelect) && mMobileBookmarks != null) {
    // ...
```

Reply:
Addressed here 067c9b454b850a75252f156f812d3bef4c54eb97